src/fields/field_fe.cc

```diff
@@ -290,11 +310,132 @@ void FieldFE<spacedim, Value>::value_list (const std::vector< Point >  &point_li
 
 
 template <int spacedim, class Value>
+void FieldFE<spacedim, Value>::init_from_input(const Input::Record &rec, const struct FieldAlgoBaseInitData& init_data) {
+	this->init_unit_conversion_coefficient(rec, init_data);
+	flags_ = init_data.flags_;
+
+
+	// read mesh, create tree
+    {
+       reader_file_ = FilePath( rec.val<FilePath>("mesh_data_file") );
+       auto reader_data = ReaderInstance::get_instance(reader_file_);
+       source_mesh_ = reader_data.mesh_;
```

Comment:
Move reading the mesh into set_time. And rather make source_mesh_ a local variable in order to free lot of memory. On the other hand we would like to keep it to reuse it when more FieldFE are interpolated from the same source mesh. But this is work for future.